app/Http/Controllers/QuizApiController.php

```diff
@@ -0,0 +1,23 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Quiz;
+use Illuminate\Http\Request;
+use Illuminate\Http\JsonResponse;
+
+class QuizApiController extends Controller
+{
+    public function all(){
+        $quizzes = Quiz::all()->makehidden(['updated_at', 'created_at']);
+        $outcome = response() -> json([
+            'message' => 'Quizzes retrieved!',
+            'data' => $quizzes,
+        ]);
+        if ($outcome == null){
+            return "Failed to retrieve quizzes...";
+        } else {
+            return $outcome;
+        }
```

Comment:
In this case, there isn't any need to have this check with two different responses.

If there are no quizes, the front-end is just expecting to receive an empty array where there would otherwise be quizzes.

So you can just do:

``` 
return response()->json([........]);
```